server-extension/src/main/java/flyway/paikkis/V2_18__link_selkokartta_to_bglayer_selector.java

```diff
@@ -0,0 +1,108 @@
+package flyway.paikkis;
+
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import fi.nls.oskari.util.JSONHelper;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+// Note to English speakers: "Selkokartta" is a background layer for cognitively impaired users
+
+public class V2_18__link_selkokartta_to_bglayer_selector implements JdbcMigration {
+    private final Logger LOG = LogFactory.getLogger(this.getClass());
+
+    private static final String EXTERNAL_ID = "base_selko";
+    private static final String PLUGIN_ID = "Oskari.mapframework.bundle.mapmodule.plugin.BackgroundLayerSelectionPlugin";
+
+    public void migrate(Connection connection) throws Exception {
+        Integer selkokarttaId = getSelkoLayer(connection);
+        if (selkokarttaId == null) {
+            LOG.warn("No Selkokartta layer found, skipping rest of migration!");
+            return;
+        }
+        updateLayerExternalId(connection, selkokarttaId, EXTERNAL_ID);
```

Comment:
Remove the external_id update it's an Oskari-feature that should be removed or improved and it's not necessary here.